spec/controllers/miq_report_controller/reports/editor_spec.rb

```diff
@@ -174,6 +174,47 @@
         end
       end
     end
+
+    def non_empty_category
+      cat = FactoryGirl.create(:classification, :name => "non_empty", :description => "Has entries")
+      cat.add_entry(:name => "foo", :description => "Foo")
+      cat.add_entry(:name => "bar", :description => "Bar")
+      cat
+    end
+
+    def empty_category
+      FactoryGirl.create(:classification, :name => "empty", :description => "Zero entries")
+    end
+
+    describe "#categories_hash" do
+      it "returns non-empty categories" do
+        non_empty_category
+        empty_category
+        expect(controller.send(:categories_hash)).to eq("non_empty" => "Has entries")
+      end
+
+      it "omits show: false categories" do
+        non_empty_category.update(:show => false)
```

Comment:
Do you that memoized `let` would be better than `def`?  IIUC the transaction is rolled back between every example, so if I use `let` in two examples it would still create it twice.

I think you mean `.update` is doing a 2nd unnecessary write.  Fixing...